src/components/widgets/edit-profile/edit-profile.component.js

```diff
@@ -0,0 +1,144 @@
+import React, { useReducer, useEffect } from 'react';
+import Button from '@material-ui/core/Button';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import EditIcon from '@material-ui/icons/Edit';
+import PropTypes from 'prop-types';
+import Autocomplete from '@material-ui/lab/Autocomplete';
+import Chip from '@material-ui/core/Chip';
+import FormControl from '@material-ui/core/FormControl';
+
+const reducer = (state, {
+  type, value, key
+}) => {
+  switch (type) {
+    case 'PROFILE_CHANGED':
+      return { ...state, ...value };
+    case 'ADD_SKILL':
+      return { ...state, skills: value ? [...state.skills, value] : [...state.skills] };
```

Comment:
In case of false branch, the new array is not needed. You can simply pass the skills array as is.